packages/frontend/src/redux/slices/tokens/index.js

```diff
@@ -110,26 +111,10 @@ const tokensSlice = createSlice({
         },
     },
     extraReducers: ((builder) => {
-        builder.addCase(fetchOwnedTokensForContract.pending, (state, { meta }) => {
-            const { accountId, contractName } = meta.arg;
-
-            set(state, ['ownedTokens', 'byAccountId', accountId, contractName, 'loading'], true);
-            set(state, ['ownedTokens', 'byAccountId', accountId, contractName, 'error'], initialErrorState);
-        });
-        builder.addCase(fetchOwnedTokensForContract.fulfilled, (state, { meta }) => {
-            const { accountId, contractName } = meta.arg;
-
-            set(state, ['ownedTokens', 'byAccountId', accountId, contractName, 'loading'], false);
-            set(state, ['ownedTokens', 'byAccountId', accountId, contractName, 'error'], initialErrorState);
-        });
-        builder.addCase(fetchOwnedTokensForContract.rejected, (state, { meta, error }) => {
-            const { accountId, contractName } = meta.arg;
-
-            set(state, ['ownedTokens', 'byAccountId', accountId, contractName, 'loading'], false);
-            set(state, ['ownedTokens', 'byAccountId', accountId, contractName, 'error'], {
-                message: error?.message || 'An error was encountered.',
-                code: error?.code
-            });
+        handleAsyncThunkStatus({
+            asyncThunk: `${SLICE_NAME}/fetchOwnedTokensForContract`,
```

Comment:
Per feedback above would prefer to use `fetchOwnedTokensForContract.typePrefix` here instead.